examples/allennlp/allennlp_simple.py

```diff
@@ -19,59 +19,66 @@
 """
 
 import os
+import random
 import shutil
 
 import allennlp
 import allennlp.data
 import allennlp.models
 import allennlp.modules
+import numpy
 import torch
 
 import optuna
+from optuna.integration import AllenNLPPruningCallback
 
 
 DEVICE = -1  # If you want to use GPU, use DEVICE = 0.
 MAX_DATA_SIZE = 3000
+MODEL_DIR = os.path.join(os.getcwd(), "result")
+TARGET_METRIC = "accuracy"
 
-DIR = os.getcwd()
-MODEL_DIR = os.path.join(DIR, "result")
 
-GLOVE_FILE_PATH = "https://s3-us-west-2.amazonaws.com/allennlp/datasets/glove/glove.6B.50d.txt.gz"
+class SubsampledDatasetReader(allennlp.data.dataset_readers.TextClassificationJsonReader):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def _read(self, datapath):
+        data = list(super()._read(datapath))
+        random.shuffle(data)
+        yield from data[:MAX_DATA_SIZE]
```

Comment:
How about slicing the dataset before shuffling it? We can use the same dataset between epochs to assess validation accuracy.

```suggestion
        data = list(super()._read(datapath))[:MAX_DATA_SIZE]
        random.shuffle(data)
        yield from data
```